bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/api/dto/clip2/Resource.java

```diff
@@ -630,60 +629,60 @@ public State getRotaryStepsLastUpdatedState(ZoneId zoneId) {
     }
 
     /**
-     * Check if the scene resource contains a 'status.active' element. If such an element is present, returns a Boolean
-     * Optional whose value depends on the value of that element, or an empty Optional if it is not.
+     * Check if the scene resource contains a 'status.active' element. Returns a Boolean if such an element is present,
+     * whose value depends on the value of that element, or null if it is not.
      *
-     * @return true, false, or empty.
+     * @return true, false, or null.
      */
-    public Optional<Boolean> getSceneActive() {
+    public @Nullable Boolean getSceneActive() {
         if (ResourceType.SCENE == getType()) {
             JsonElement status = this.status;
             if (Objects.nonNull(status) && status.isJsonObject()) {
                 JsonElement active = ((JsonObject) status).get("active");
                 if (Objects.nonNull(active) && active.isJsonPrimitive()) {
-                    return Optional.of(!"inactive".equalsIgnoreCase(active.getAsString()));
+                    return !"inactive".equalsIgnoreCase(active.getAsString());
                 }
             }
         }
-        return Optional.empty();
+        return null;
     }
 
     /**
-     * If the getSceneActive() optional result is empty return 'UnDefType.NULL'. Otherwise if the optional result is
-     * present and 'true' (i.e. the scene is active) return the scene name. Or finally (the optional result is present
-     * and 'false') return 'UnDefType.UNDEF'.
+     * If the getSceneActive() result is null or false return 'UnDefType.NULL'. Otherwise if the result is
+     * 'true' (i.e. the scene is active) return the scene name.
      *
-     * @return either 'UnDefType.NULL', a StringType containing the (active) scene name, or 'UnDefType.UNDEF'.
+     * @return either a StringType containing the (active) scene name, or 'UnDefType.UNDEF'.
      */
     public State getSceneState() {
-        return getSceneActive().map(a -> a ? new StringType(getName()) : UnDefType.UNDEF).orElse(UnDefType.NULL);
+        Boolean sceneActive = getSceneActive();
```

Comment:
This code is wrong. Please read the original JavaDoc. Possible return results are UnDefType.UNDEF, UnDefType.NULL, or StringType()..

Reply:
i.e. it must return the following:

- if the optional has a content, and the content is true => StringType
- if the optional has a content, and the content is false => UndefType.UNDEF
- if the optional has no content => UndefType.NULL

Reply:
Fixed on both places, thanks!

Reply:
Can you point me in the right direction, because i already adjusted the javadoc in this commit: https://github.com/openhab/openhab-addons/pull/17293/commits/05cdac1c23333209285672165f8677e3d3e9f731

Reply:
i.e. /* @return either 'UnDefType.NULL', a StringType containing the (active) scene name, or 'UnDefType.UNDEF'.

Reply:
Thanks, i adjusted just one of two. Now all should be fine.